### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

## Changes

* add more info about components integration with Sessions
* update docs with new `AdyenContext` and `DropInComponent.Configuration` public interface
* format API Reference.md
* restructure Adyen.md
* add references to the docs

## Demo

Screenshot 2022-07-18 at 12 08 40

### CODE REVIEW

File: Adyen.docc/Components.md [hunk 1]
```diff
@@ -26,3 +26,154 @@ In order to have more flexibility over the checkout flow, you can use our Compon
...
+    case let .success(session):
+        // store the session object
+        self?.session = session
+    case let .failure(error):
+        // handle the error
+    }
+}
+```
+
+Fetch `paymentMethod` from `paymentMethods` response or create one manually.
```
Comment (mohammedDehairy) [inline Adyen.docc/Components.md]: thats not correct, they should use `session.sessionContext.paymentMethods` :-)

File: Adyen.docc/Components.md [hunk 2]
```diff
@@ -26,3 +26,154 @@ In order to have more flexibility over the checkout flow, you can use our Compon
...
+                      PKPaymentSummaryItem(label: "Item B", amount: 25, type: .final),
+                      PKPaymentSummaryItem(label: "My Company", amount: 100, type: .final)
+                   ]
+let applePayment = try ApplePayPayment(countryCode: "US",
+                                       currencyCode: "USD",
+                                       summaryItems: summaryItems)
+
+let dropInConfiguration = DropInComponent.Configuration()
+dropInConfiguration.applePay = .init(payment: applePayment,
+                               merchantIdentifier: "merchant.com.adyen.MY_MERCHANT_ID")
```
Comment (mohammedDehairy) [inline Adyen.docc/Components.md]: why is dropIn mentioned in a section about individual components?

Comment (descorp) [reply Adyen.docc/Components.md]: 👍

File: Adyen.docc/Components.md (hunk 1 above)
Comment (descorp) [reply Adyen.docc/Components.md]: 👍

**Review by mohammedDehairy — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

**Pull request merged**